src/redis-helper.ts

```diff
@@ -199,12 +270,17 @@ export async function getNewestTracksForArtist(redisClient: redis.RedisClient,
     const newestTracksFromRedis: RedisTrack[]     = [];
     let   newestTracksFromSpotify: SpotifyTrack[] = [];
 
-    if (!artist.newest_track_ids || artist.newest_track_ids.length === 0) {
-        // We've never gotten albums and saved their ids for this artist, need to call spotify for albums,
-        // save their ids for this artist, save the albums themselves, and (EXTRA STEP COMPARED TO TOP TRACKS) get the
-        // tracks from the most recent album(s). We cache up to 50 album IDs per artist (max recieved in spotify query)
-        // and albums themselves, but only return the number requested
-        console.log(`No newest track ids for spotify artist ${
+    const lastUpdated = await getArtistNewestTracksLastUpdated(redisClient, artist.id);
+
+    console.log(`Last updated newest tracks: ${lastUpdated}`);
+
+    if (!artist.newest_track_ids || artist.newest_track_ids.length === 0 || !lastUpdated ||
+        lastUpdated < (new Date(Date.now() - constants.daysInMsBeforeArtistTracksRefresh))) {
+        // We've never gotten albums and saved their ids for this artist, or haven't done it for a while, need to call
+        // spotify for albums, save their ids for this artist, save the albums themselves, and (EXTRA STEP COMPARED TO
+        // TOP TRACKS) get the tracks from the most recent album(s). We cache up to 50 album IDs per artist (max
+        // recieved in spotify query) and albums themselves, but only return the number requested
+        console.log(`No/out of date newest track ids for spotify artist ${
             artist.id}, getting albums from spot, saving to redis albums, then getting tracks from most recent album and saving each track`);
         // Technically we could optimize a bit here by checking to see if we already have the albums and skip a query.
         // We also don't handle the case of if the artist releases a new album after we've already newest tracks from
```

Comment:
You can chop this sentence since it wouldn't be true anymore

Reply:
Done